xbmc/windowing/wayland/WinEventsWayland.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ *      Copyright (C) 2017 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "WinEventsWayland.h"
+
+#include "Application.h"
+#include "utils/log.h"
+
+using namespace KODI::WINDOWING::WAYLAND;
+
+namespace
+{
+wayland::display_t* g_WlDisplay = nullptr;
+}
+
+void CWinEventsWayland::SetDisplay(wayland::display_t* display)
+{
+  g_WlDisplay = display;
+}
+
+size_t CWinEventsWayland::GetQueueSize()
+{
+  // FIXME Where is this function used?
+  throw std::runtime_error("Not implemented");
+}
+
+// FIXME Why is the return value of this ignored everywhere?
```

Comment:
MessagePump is supposed to die. In some future version rendering should run on its own thread and this thread can send events to application.